packages/dd-trace/src/appsec/rasp.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+const { storage } = require('../../../datadog-core')
+const addresses = require('./addresses')
+const { httpClientRequestStart } = require('./channels')
+const waf = require('./waf')
+
+function enable () {
+  httpClientRequestStart.subscribe(analyzeSsrf)
+}
+
+function disable () {
+  httpClientRequestStart.unsubscribe(analyzeSsrf)
```

Comment:
if you take a look at appsec/index.js `disable()`, you'll see I wrote `if (chan.unsubscribe) chan.unsubscribe()` because there was a bug in node core where the unsubscribe method wouldn't exist if there was already no subscriber. I patched the bug in node a while ago but i'm not sure if it's still present in versions we support or not. Could be worth checking if we still need it ? and if you don't have time to check now, then i'd say let's put the safety net just in case

Reply:
the fix in question: https://github.com/nodejs/node/pull/40433/files#diff-582bebd268f1b390f827a3328ecca51c09e7f2befb807249ebf355d19472bc1cR84-R86

Reply:
it's an easy check that we can do without any cost... but I though that one of the reasons to use `dc-polyfill` was to be safe of these cases.